presto-parser/src/main/java/io/prestosql/sql/tree/GroupingSets.java

```diff
@@ -99,4 +99,10 @@ public String toString()
                 .add("sets", sets)
                 .toString();
     }
+
+    @Override
+    public boolean shallowEquals(Node other)
+    {
+        return sameClass(this, other);
```

Comment:
the field `List<List<Expression>> sets` is neither compared in `shallowEquals()` nor are the expressions returned in `getChildren()`. 
I think that the number and sizes of the nested lists should be compared in `shallowEquals`.
In `getChildren()` a concatenated list should be returned.